pkg/api/register.go

```diff
@@ -111,6 +111,15 @@ func AddToScheme(scheme *runtime.Scheme) {
 		&unversioned.APIGroup{},
 		&unversioned.APIResourceList{},
 	)
+
+	// Types registered here will be kept unversioned when they are encoded.
+	// It's for backward compatibility with release 1.1. clients.
+	Scheme.AddToKeepUnverioned(
```

Comment:
Agree, waiting for v2 might take too long time. @smarterclayton, any suggestion on how to implement the different behavior for different groups? I can only think of very hacky implementation.